### PR TITLE
Feature/update to polkadot v0 9 12

### PR DESCRIPTION
**Pull request by MRamanenkau**

### CODE REVIEW

**Review by AndreiNavoichyk — DISMISSED** [review]

File: bin/node-template/shell.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+let
```
Comment (AndreiNavoichyk) [inline bin/node-template/shell.nix:1]: We don't need `node-template`

File: bin/node/runtime/src/lib.rs [hunk 2]
```diff
@@ -503,6 +500,7 @@ parameter_types! {
 	pub const SlashDeferDuration: pallet_staking::EraIndex = 2;
 	pub const RewardCurve: &'static PiecewiseLinear<'static> = &REWARD_CURVE;
 	pub const MaxNominatorRewardedPerValidator: u32 = 256;
+	pub const OffendingValidatorsThreshold: Perbill = Perbill::from_percent(17);
```
Comment (AndreiNavoichyk) [inline bin/node/runtime/src/lib.rs:503]: What is this?

File: bin/node-template/shell.nix (hunk 1 above)
Comment (MRamanenkau) [reply bin/node-template/shell.nix:1]: [Removed in v0.9.16](https://github.com/Cerebellum-Network/pos-network-node/pull/165/commits/574a020bcef14992638e944489806ebc59b1cb83)

File: bin/node/runtime/src/lib.rs (hunk 2 above)
Comment (MRamanenkau) [reply bin/node/runtime/src/lib.rs:503]: [The value moved from DisabledValidatorsThreshold variable ](https://github.com/paritytech/substrate/pull/9448/files#diff-67684005af418e25ff88c2ae5b520f0c040371f1d817e03a3652e76b9485224aL463).
Below the line there is answer why exactly 17%

**Review by AndreiNavoichyk — APPROVED** [review]

**Pull request merged**